### PR TITLE
CB-11878 When account has DATAHUB_SCALING_STEP_SIZE entitlement, scal…

### PR DESCRIPTION
**Pull request by smaniraju**
…e up is allowed in steps of 200.

See detailed description in the commit message.

### CODE REVIEW

File: autoscale-api/src/main/java/com/sequenceiq/periscope/api/model/LoadAlertConfigurationRequest.java [hunk 1]
```diff
@@ -17,14 +17,14 @@
 public class LoadAlertConfigurationRequest implements Json {
     @ApiModelProperty(LoadAlertJsonProperties.LOAD_ALERT_CONFIGURATION_MIN_RESOUCE_VALUE)
     @Min(value = 0)
-    @Max(value = 200)
+    @Max(value = 500)
     @Digits(fraction = 0, integer = 3)
     @NotNull
```
Comment (smaniraju) [inline autoscale-api/src/main/java/com/sequenceiq/periscope/api/model/LoadAlertConfigurationRequest.java:22]: CB now supports manual scaling up-to 500 also, hence updating autoscaling limits also.

File: cloud-api/src/main/java/com/sequenceiq/cloudbreak/common/ScalingHardLimitsService.java [hunk 2]
```diff
@@ -7,14 +7,23 @@
 public class ScalingHardLimitsService {
     private static final int DEFAULT_UPSCALING_MAXIMUM_NODE_COUNT = 100;
 
+    private static final int DEFAULT_UPSCALING_NODE_COUNT_WHEN_SCALING_STEP_ENTITLEMENT = 200;
```
Comment (sidseth) [inline cloud-api/src/main/java/com/sequenceiq/cloudbreak/common/ScalingHardLimitsService.java:10]: Should this be configurable as well?

File: autoscale-api/src/main/java/com/sequenceiq/periscope/api/model/LoadAlertConfigurationRequest.java [hunk 3]
```diff
@@ -17,14 +17,14 @@
 public class LoadAlertConfigurationRequest implements Json {
     @ApiModelProperty(LoadAlertJsonProperties.LOAD_ALERT_CONFIGURATION_MIN_RESOUCE_VALUE)
     @Min(value = 0)
-    @Max(value = 200)
+    @Max(value = 500)
```
Comment (sidseth) [inline autoscale-api/src/main/java/com/sequenceiq/periscope/api/model/LoadAlertConfigurationRequest.java:20]: Where does the 500 come from? CB config / hard-coded?

File: autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/ScalingRequest.java [hunk 4]
```diff
@@ -104,10 +108,11 @@ public void run() {
 
     private void scaleUp(int scalingAdjustment, int totalNodes) {
         metricService.incrementMetricCounter(MetricType.CLUSTER_UPSCALE_TRIGGERED);
-        if (scalingHardLimitsService.isViolatingMaxUpscaleStepInNodeCount(scalingAdjustment)) {
-            LOGGER.debug("Upscale requested for '{}' nodes. Upscaling with the maximum allowed of '{}' node(s)",
-                    scalingAdjustment, scalingHardLimitsService.getMaxUpscaleStepInNodeCount());
-            scalingAdjustment = scalingHardLimitsService.getMaxUpscaleStepInNodeCount();
+        int maxScaleUpStepAllowed = entitlementValidationService.scalingStepEntitlementEnabled(cluster.getClusterPertain().getTenant()) ?
```
Comment (sidseth) [inline autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/ScalingRequest.java:111]: Here, and in Periscope ... can this be changed to test against the default first. If that fails, check with the entitlement service. (Avoid making calls to the entitlement service for every such request)

File: auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/EntitlementService.java [hunk 5]
```diff
@@ -242,6 +243,10 @@ public boolean gcpAutoScalingEnabled(String accountId) {
         return isEntitlementRegistered(accountId, DATAHUB_GCP_AUTOSCALING);
     }
 
+    public boolean dataHubScalingStepSizeEnabled(String accountId) {
+        return isEntitlementRegistered(accountId, CDP_DATAHUB_SCALING_STEP_SIZE);
```
Comment (sidseth) [inline auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/EntitlementService.java:247]: Maybe rename to CDP_DATAHUB_LARGE_SCALING_STEP_SIZE

**Pull request closed without merging**